DeDriftAndResample/MSMRemoveGroupDrift.sh

```diff
@@ -4,488 +4,326 @@
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
 
-EOF
-}
+# Load function libraries
+source "${HCPPIPEDIR}/global/scripts/debug.shlib" "$@"         # Debugging functions; also sources log.shlib
+source ${HCPPIPEDIR}/global/scripts/opts.shlib                 # Command line option functions
 
-# ------------------------------------------------------------------------------
-#  Get the command line options for this script.
-# ------------------------------------------------------------------------------
+opts_SetScriptDescription "MSM Remove Group Drift - Compute Group Registration Drift"
 
-get_options()
-{
-	local arguments=($@)
-
-	# initialize global output variables
-	unset p_StudyFolder     # StudyFolder
-	unset p_Subjlist             # Subjlist
-	unset p_CommonFolder            # CommonFolder
-	unset p_GroupAverageName       # GroupAverageName
-	unset p_InputRegName  # InRegName
-	unset p_TargetRegName # TargetRegName
-	unset p_RegName                # RegName
-	unset p_HighResMesh            # HighResMesh
-	unset p_LowResMesh             # LowResMesh
-
-	# set default values
-
-	# parse arguments
-	local num_args=${#arguments[@]}
-	local argument
-	local index
-
-    for ((index = 0; index < num_args; ++index))
-    do
-		argument=${arguments[index]}
-
-		case ${argument} in
-			--help)
-				show_usage
-				exit 0
-				;;
-			--path=*)
-				p_StudyFolder=${argument#*=}
-				;;
-			--study-folder=*)
-				p_StudyFolder=${argument#*=}
-				;;
-			--subject-list=*)
-				p_Subjlist=${argument#*=}
-				;;
-			--common-folder=*)
-				p_CommonFolder=${argument#*=}
-				;;
-			--group-average-name=*)
-				p_GroupAverageName=${argument#*=}
-				;;
-			--input-registration-name=*)
-				p_InputRegName=${argument#*=}
-				;;
-			--target-registration-name=*)
-				p_TargetRegName=${argument#*=}
-				;;
-			--registration-name=*)
-				p_RegName=${argument#*=}
-				;;
-			--high-res-mesh=*)
-				p_HighResMesh=${argument#*=}
-				;;
-			--low-res-mesh=*)
-				p_LowResMesh=${argument#*=}
-				;;
-			*)
-				show_usage
-				log_Err_Abort "unrecognized option: ${argument}"
-				;;
-		esac
-	done
+#ARE THESE DESCRIPTIONS OK?????
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "path to study folder"
 
-	local error_count=0
-	# check required parameters
-	if [ -z "${p_StudyFolder}" ]; then
-		echo "ERROR: path to study folder (--path= or --study-folder=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_StudyFolder: ${p_StudyFolder}"
-	fi
+opts_AddMandatory '--subject-list' 'Subjlist' 'list' "@ delimited list of subject ids"
 
-	if [ -z "${p_Subjlist}" ]; then
-		echo "ERROR: subject ID list (--subject-list=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "p_Subjlist: ${p_Subjlist}"
-	fi
+opts_AddMandatory '--common-folder' 'CommonFolder' 'path' "path to study folder"
```

Comment:
@glasserm This description is wrong, but I don't know what this argument is.

Reply:
It is the group average folder being used for the dedrifting.

Reply:
Should we change the primary name for this switch to `--group-folder` (and put `--common-folder` as a compatibility flag)?

Reply:
Sure, CommonFolder is an older terminology I used.

Reply:
Done